src/hash_tree.rs

```diff
@@ -351,21 +354,21 @@ impl<'a> HashTreeNode<'a> {
         match self {
             // If this node is a labeled node, check for the name. This assume a
             HashTreeNode::Labeled(l, node) => match label.cmp(l) {
-                std::cmp::Ordering::Greater => LookupLabelResult::Continue,
+                std::cmp::Ordering::Greater => LookupLabelResult::Greater,
                 std::cmp::Ordering::Equal => LookupLabelResult::Found(node.as_ref()),
                 // If this node has a smaller label than the one we're looking for, shortcut
                 // out of this search (sorted tree), we looked too far.
-                std::cmp::Ordering::Less => LookupLabelResult::Absent,
+                std::cmp::Ordering::Less => LookupLabelResult::Less,
             },
             HashTreeNode::Fork(nodes) => {
                 let left_label = nodes.0.lookup_label(label);
                 match left_label {
                     // On continue or unknown, look on the right side of the fork.
                     // If it cannot be found on the right, return Unknown though.
-                    LookupLabelResult::Continue | LookupLabelResult::Unknown => {
+                    LookupLabelResult::Greater | LookupLabelResult::Unknown => {
                         let right_label = nodes.1.lookup_label(label);
                         match right_label {
-                            LookupLabelResult::Absent => {
+                            LookupLabelResult::Less => {
                                 if matches!(left_label, LookupLabelResult::Unknown) {
```

Comment:
This code looks somewhat weird: we first match both `Greater` and `Unknown`, and then we're trying to figure out which one it was. I'd just split the "or" clause; it looks cleaner, IMO.